src/xtopcom/xvledger/src/xvblock.cpp

```diff
@@ -1065,14 +1076,27 @@ namespace top
             }
             
             const std::string data_to_sign((const char*)stream.data(),stream.size());
-            return hash(data_to_sign);
+            //if m_modified_count != 0, rebuild hash 
+            if (check_modified_change() > 0) {
```

Comment:
does this mean something has changed after the checking in the beginning, which will invalid the calculated data_to_sign?